api/v1alpha1/fenceagentsremediation_params.go

```diff
@@ -0,0 +1,416 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"maps"
+	"slices"
+
+	commonAnnotations "github.com/medik8s/common/pkg/annotations"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilErrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/medik8s/fence-agents-remediation/pkg/template"
+	"github.com/medik8s/fence-agents-remediation/pkg/validation"
+)
+
+const (
+	parameterRebootActionValue     = "reboot"
+	parameterOffActionValue        = "off"
+	actionName                     = "action"
+	parameterActionName            = "--" + actionName
+	errorParamDefinedMultipleTimes = "invalid multiple definition of FAR parameter, parameter name: %s"
+	errorMissingParams             = "invalid template: mandatory parameters are missing"
+	ErrorUnsupportedAction         = "FAR doesn't support any other action than reboot or off"
```

Comment:
NIT: Maybe add an apostrophe to distinguish that this word (reboot/off) is an actual field value?

```suggestion
	ErrorUnsupportedAction         = "FAR doesn't support any other action than `reboot` or `off`"
```